torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -196,6 +208,34 @@ class TensorExprFuser {
       updateQueue(fusion_group, queue, visited_nodes);
     }
 
+    // Update typeinfo for inputs and outputs of the fusion group
```

Comment:
So, the reason we're not updating the typeinfo map until after we've already exhausted the queue is because we never expect to successfully merge two Fusion groups together. I think this is always the case but I'm not totally sure. It might be worth updating the fusion group outputs as part of updateTypeinfoMapWithVmap instead of deferring it till after

Reply:
based on conversation today, I guess we do expect to merge two fusion groups together

Reply:
That is correct observation, I'm going to revisit this part as I'll be working on merging adjacent fusion groups.